Classes/Domain/Collection/Collection.php

```diff
@@ -198,6 +198,22 @@ public function first()
         return current($this->collection);
     }
 
+    /**
+     * Reset array pointer.
+     *
+     * @return array|null
+     */
+    public function rewind(): ?array
+    {
+        if (empty($this->collection)) {
+            return null;
```

Comment:
As we return an optional array here. Would it not be better to have array as return type and return an empty array if the collection is empty?